packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,384 @@
+							node={node}
+							setEditing={setEditing}
+							submitChange={submitChange}
+						/>
+					);
+					break;
+				case "number":
+					component = (
+						<EditableNumberComponent
+							node={node}
+							setEditing={setEditing}
+							submitChange={submitChange}
+						/>
+					);
+					break;
+				case "boolean":
+					component = (
+						<EditableBooleanComponent node={node} submitChange={submitChange} />
+					);
+					break;
+				case "undefined":
+					break;
+
+				case "null":
+					break;
+
+				default:
+			}
+
+			return component;
+		},
+		[node, submitChange],
+	);
+
+	/**
+	 * Background updates to keep the UI in sync with the data
+	 */
+	React.useEffect(() => {
+		if (editing === false) {
+			setType(typeof node.value);
+			// setLocalData(node.value);
+			setEditingComponent(selectEditUI(typeof node.value));
+		}
+	}, [node.value, editing, selectEditUI]);
+
+	/**
+	 * Updates {@link "type"} and {@link editingComponent} based on the option selected from the dropdown
+	 * @param data - The option selected from the dropdown
+	 */
+	const onOptionSelect: DropdownProps["onOptionSelect"] = (event, data) => {
+		setType(data.optionText ?? "undefined");
+		setEditingComponent(selectEditUI(data.optionText ?? "undefined"));
+		if (data.optionText === "undefined") {
+			submitChange(undefined);
+		}
+
+		if (data.optionText === " null") {
+			// We need to support users waiting to use "null" as a value
+			// eslint-disable-next-line unicorn/no-null
+			submitChange(null);
+		}
+	};
+
+	const options = node.editProps?.editTypes;
+	return options === undefined ? (
+		<TreeHeader
+			label={label}
+			nodeTypeMetadata={node.typeMetadata}
+			inlineValue={String(node.value)}
+		/>
+	) : (
+		<div
+			style={
+				editing
+					? {
+							border: "2px solid",
+							borderColor: tokens.colorBrandStroke1,
+							display: "flex",
+					  }
+					: { display: "flex" }
+			}
+			onBlur={onBlur}
+			onFocus={onFocus}
+			onClick={(event): void => event?.preventDefault()}
+		>
+			<TreeHeader label={label} nodeTypeMetadata={node.typeMetadata} />
+			<Dropdown
+				size="small"
+				style={{ minWidth: "10px" }}
+				onOptionSelect={onOptionSelect}
+				value={type}
+				selectedOptions={[type]}
+			>
+				{options.map((option) => (
+					<Option key={option}>{option}</Option>
+				))}
+			</Dropdown>
+			{editingComponent}
+		</div>
+	);
+}
+
+/**
+ * Input to {@link EditBooleanComponent}
+ */
+interface EditableBooleanComponentProps {
+	node: FluidObjectValueNode;
+	submitChange: (data: Serializable<unknown> | Primitive) => void;
+}
+
+/**
+ * Component which allows for boolean editing
+ */
+function EditableBooleanComponent(props: EditableBooleanComponentProps): React.ReactElement {
+	const { node, submitChange } = props;
+
+	/**
+	 * Auxiliary function to make boolean editing buttons
+	 * @param isTrue - Determines if it should be a true or false button
+	 * @returns - 2 buttons, one for setting value to false and one for true
```

Comment:
It's unintuitive, since `@param` blocks require a `-` between tag and docs, but `@returns` (and most others) don't want a `-`.
```suggestion
	 * @returns 2 buttons, one for setting value to false and one for true
```

Reply:
Basically what you said is what made me assume it was with the hyphen